pkg/sbom/attestation/attestation.go

```diff
@@ -0,0 +1,30 @@
+package attestation
+
+import (
+	"bytes"
+	"io"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/attestation"
+	"github.com/aquasecurity/trivy/pkg/sbom"
+)
+
+type Unmarshaler struct {
+	predicateUnmarshaler sbom.Unmarshaler
+}
+
+func (u Unmarshaler) Unmarshal(r io.Reader) (sbom.SBOM, error) {
+	attest, err := attestation.Decode(r)
+	if err != nil {
+		return sbom.SBOM{}, xerrors.Errorf("failed to decode attestation: %w", err)
+	}
+
+	return u.predicateUnmarshaler.Unmarshal(bytes.NewReader(attest.Predicate.Data))
```

Comment:
Could you wrap the error if it is returned? It adds stack trace and helps us debug.

Reply:
Done